analytics-core/src/main/scala/org/ekstep/analytics/framework/util/MergeUtil.scala

```diff
@@ -0,0 +1,109 @@
+package org.ekstep.analytics.framework.util
+
+import org.apache.commons.io.FilenameUtils
+import org.apache.hadoop.fs.Path
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.functions.{col, unix_timestamp, _}
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.ekstep.analytics.framework.{FrameworkContext, MergeScriptConfig, StorageConfig}
+import org.joda.time
+import org.joda.time.format.DateTimeFormat
+import org.ekstep.analytics.framework.util.DatasetUtil.extensions
+
+class MergeUtil {
+
+    def mergeFile(mergeConfig: MergeScriptConfig)(implicit sc: SparkContext, fc: FrameworkContext): Unit = {
+        implicit val sqlContext = new SQLContext(sc)
+        mergeConfig.merge.files.foreach(filePaths => {
+            val path = new Path(filePaths("reportPath"))
+            val mergeResult:(DataFrame, StorageConfig) = mergeConfig.`type`.toLowerCase() match {
+                case "local" =>
+                    val deltaDF = sqlContext.read.options(Map("header" -> "true")).csv(filePaths("deltaPath"))
+                    val reportDF = sqlContext.read.options(Map("header" -> "true")).csv(filePaths("reportPath"))
+                    (mergeReport(deltaDF, reportDF, mergeConfig),StorageConfig(mergeConfig.`type`, null, FilenameUtils.getFullPathNoEndSeparator(filePaths("reportPath"))))
+                case "azure" =>
+                    val deltaDF = downloadAzureFile(filePaths("deltaPath"),
+                        mergeConfig.deltaFileAccess.getOrElse(true), mergeConfig.container)
+
+                    val reportDF = downloadAzureFile(filePaths("reportPath"), mergeConfig.reportFileAccess.getOrElse(true),
+                        mergeConfig.postContainer.getOrElse("reports"))
+                    (mergeReport(deltaDF, reportDF, mergeConfig),StorageConfig(mergeConfig.`type`, mergeConfig.postContainer.get, path.getParent.getName))
+                case _ =>
+                    throw new Exception("Merge type unknown");
+            }
+            val mergeDF= mergeResult._1
+            mergeDF.saveToBlobStore(mergeResult._2, "csv", FilenameUtils.removeExtension(path.getName), Option(Map("header" -> "true", "mode" -> "overwrite")), None)
+            mergeDF.saveToBlobStore(mergeResult._2, "json", FilenameUtils.removeExtension(path.getName), Option(Map("header" -> "true", "mode" -> "overwrite")), None)
+        })
+    }
+
+
+    def mergeReport(delta: DataFrame, reportDF: DataFrame, mergeConfig: MergeScriptConfig): DataFrame = {
+
+        if (mergeConfig.rollup > 0) {
+            val defaultFormat = "dd-MM-yyyy"
+            val reportDfColumns = reportDF.columns
```

Comment:
We would need an append mode where we just append the delta data to the existing data on few columns for the same on the same rollupCol? For ex: adding data to a cumulative report (yearly or monthly or weekly cumulative files)